### DISCUSSION TITLE
Sandboxed program (Rainmeter) can't set z-position?

### DISCUSSION THREAD

**Discussion by e-t-l** [sandboxie-plus/Sandboxie#4221] [General]
[Rainmeter](https://github.com/rainmeter/rainmeter) is a "desktop skin" customization program that can set itself as always-on-the-desktop so that it is unaffected by e.g. Win+D commands that minimize all windows (I believe the relevant C++ is `SetWindowPos(hwndTT, HWND_TOPMOST, 0, 0, 0, 0, SWP_NOMOVE | SWP_NOSIZE | SWP_NOACTIVATE);`). 

When Rainmeter is sandboxed in a standard yellow sandbox, this window z-position is ignored, and Rainmeter's desktop skins can get minimized. 

Is this something that can be fixed through the sbie config, or is it a bug?

**Comment by offhub** [comment]
Try this:
```ini
OpenWinClass=Rainmeter.exe,*
```

**Reply by e-t-l** [reply]
This works, although doesn't `OpenWinClass=*` open up all outside processes to the sandboxed one, thereby reducing security/privacy?
As a mitigation, I also added
```
HideNonSystemProcesses=y
HideOtherBoxes=y
Template=BlockAccessWMI
```
Is that reasonable?

**Reply by e-t-l** [reply]
`OpenWinClass=Rainmeter.exe,*` no longer works, by the way. Not sure if it's due to a Windows update or a Sandboxie update

**Reply by offhub** [reply]
Bug:

https://www.rainmeter.net/changes-4.5/

> April 11, 2025 - 4.5.22 - Revision 3835
>     FixedRainmeter: Fixed a problem with Windows 11 version 24H2 or later,
>     that didn't allow Rainmeter skins to remain visible when the WIN-D (Show Desktop) function of Windows was used.
> 



When I press Win+D it disappears for a short while but then reappears. As far as I remember it used to be like this. (SP v1.15.12 and Win 10)

https://github.com/user-attachments/assets/6c8a0131-ffb6-45ac-854b-2cc68c8ee429

**Reply by e-t-l** [reply]
Thanks, I hadn't seen the rainmeter update, ~~that fixed it~~

**Reply by e-t-l** [reply]
Hmm, doesn't seem like it's actually fixed for me, even after updating Rainmeter. It's strange, if Rainmeter is running normally (i.e. just from the tray), then Win+D will make it minimize. However, if I open the Rainmeter Manager window, *then* I get the behavior seen in your screen recording. It still behaves like that if I minimize the Rainmeter Manager, but as soon as I close that window, the rainmeter skins are susceptible to being minimized again. 

Tell me, when you recorded that video, did you have the windowed "Manage Rainmeter" process running? And do you have rainmeter installed inside or outside the sandbox?

**Reply by offhub** [reply]
I tested Rainmeter both inside and outside the sandbox, as well as with the 'Manage Rainmeter' window open and closed. As long as `OpenWinClass=Rainmeter.exe,*` is set in the configuration, the behavior remains the same. Just make sure there are no leftover executables from an old version in the sandbox location. (Windows 11 24H2 VM)
